tests/lax_numpy_test.py

```diff
@@ -3931,19 +3931,43 @@ def testAstypeBool(self, from_dtype, use_method, to_dtype='bool'):
     self._CompileAndCheck(jnp_op, args_maker)
 
   @jtu.sample_product(
-    change_dtype=[True, False],
+    [dict(dtype=dtype, new_dtype=new_dtype)
+      for dtype in all_dtypes
+      for new_dtype in (
+        complex_dtypes
+        if np.issubdtype(dtype, np.complexfloating)
+        else all_dtypes
+      )
+    ],
+    shape=array_shapes,
     copy=[True, False],
+    device_type=[None, "single", "shard"],
   )
-  def testAstypeCopy(self, change_dtype, copy):
-    dtype = 'float32' if change_dtype else 'int32'
-    expect_copy = change_dtype or copy
-    x = jnp.arange(5, dtype='int32')
-    y = x.astype(dtype, copy=copy)
+  @jtu.run_on_devices("gpu")
+  def testAstypePlacement(self, shape, dtype, new_dtype, copy, device_type):
+    rng = jtu.rand_default(self.rng())
+    x = jnp.asarray(rng(shape, dtype))
+
+    if device_type is None:
+      device = None
+      expected_sharding = x.sharding
+    elif device_type == "single":
+      device = jax.devices("cpu")[0]
+      expected_sharding = SingleDeviceSharding(device)
+    else:
+      global_mesh = jtu.create_global_mesh((4, 2), ('x', 'y'))
+      device = jax.sharding.NamedSharding(global_mesh, P('x', 'y'))
```

Comment:
It would be nice to try other PartitionSpecs too (can you parameterize the test on that too)